Tests/Editor/SettingsAnalysisTests.cs

```diff
@@ -342,5 +342,46 @@ public void SettingsAnalysis_FogStripping_IsNotReported(FogStripping fogModeStri
 
             serializedObject.ApplyModifiedProperties();
         }
+
+        [Test]
+        [TestCase(Il2CppCompilerConfiguration.Debug)]
+        [TestCase(Il2CppCompilerConfiguration.Master)]
+        public void SettingsAnalysis_IL2CPP_Compiler_Configuration_IsReported(
+            Il2CppCompilerConfiguration il2CppCompilerConfiguration)
+        {
+            var compilerConfiguration = PlayerSettings.GetIl2CppCompilerConfiguration(EditorUserBuildSettings.selectedBuildTargetGroup);
+            PlayerSettings.SetIl2CppCompilerConfiguration(EditorUserBuildSettings.selectedBuildTargetGroup, il2CppCompilerConfiguration);
+
+            ProjectIssue[] issues = null;
+
+            switch (il2CppCompilerConfiguration)
+            {
+                case Il2CppCompilerConfiguration.Master:
+                    issues = Analyze(IssueCategory.ProjectSetting, i => i.descriptor.id.Equals("PAS1004"));
+                    break;
+                case  Il2CppCompilerConfiguration.Debug:
+                    issues = Analyze(IssueCategory.ProjectSetting, i => i.descriptor.id.Equals("PAS1005"));
+                    break;
+            }
+
+            var playerSettingIssue = issues.FirstOrDefault();
+
+            Assert.NotNull(playerSettingIssue);
+
+            PlayerSettings.SetIl2CppCompilerConfiguration(EditorUserBuildSettings.selectedBuildTargetGroup, compilerConfiguration);
+        }
+
```

Comment:
could you please add two additional tests:
- _SettingsAnalysis_Il2CppCompilerConfigurationRelease_IsNotReported_
- _SettingsAnalysis_Il2CppCompilerConfigurationMaster_ScriptingBackendMono_IsNotReported_